include/itkDICOMImage.h

```diff
@@ -0,0 +1,163 @@
+/*=========================================================================
+ *
+ *  Copyright Insight Software Consortium
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+#ifndef itkDICOMImage_h
+#define itkDICOMImage_h
+
+#include "itkImage.h"
+
+namespace itk
+{
+/** \class DICOMImage
+ *  \brief Templated n-dimensional image class.
```

Comment:
```
Templated n-dimensional image class representing a DICOM modality.

This class will implement its own CopyInformation() methods that copies the modality specific 
DICOM tag information. Filters will have an opportunity to adjust this information as required in 
their GenerateOutputInformation(). This will relative efficient and provide a mechanism to generate 
correct metadata.

\note Due to the implementation details of itk::MetaDataDictionary (copying it through the pipeline 
will require a lot of memory and computational overhead) and considering filters may make changes 
that require changes to the DICOM tags (intensity range, or spacing related tags), the chosen 
approach is to implement itk::DICOMImage class that inherits from itk::Image, and has a specialization
 from the different image modalities. 
```

Adapted from a prior comment from @thewtex